mas_arena/agents/EvoMAC.py

```diff
@@ -0,0 +1,791 @@
+import time
+import json
+import os
+import asyncio
+import re
+import tempfile
+import subprocess
+import sys
+from collections import defaultdict, deque
+from dotenv import load_dotenv
+from dataclasses import dataclass
+from typing import Dict, TypedDict, Any, List
+
+from pydantic import BaseModel
+from langchain_openai import ChatOpenAI
+from langchain_core.messages import SystemMessage, HumanMessage, AIMessage
+from mas_arena.agents.base import AgentSystem, AgentSystemRegistry
+from mas_arena.agents.format_prompts import get_format_prompt
+
+# Load environment variables
+load_dotenv()
+
+# change
+
+# Prompt Templates for EvoMAC - Organized by role/agent similar to MetaGPT
+PROMPTS = {
```

Comment:
This implementation is not compatible with the tool usage. Please check the `extending.md`.